### PR TITLE
add ecspi support

### PR DESCRIPTION
**Pull request by grembeter**
All three ECSPI engines are in use:
 - ECSPI1 - Inclinometer
 - ECSPI2 - UHF
 - ECSPI3 - IMU

"spidev" approach showed good results on proto boards.

### CODE REVIEW

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 1]
```diff
@@ -439,6 +471,20 @@
                >;
 	};
 
+	pinctrl_ecspi3: ecspi3grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_UART1_RXD_ECSPI3_SCLK              0x116
+			MX8MM_IOMUXC_UART1_TXD_ECSPI3_MOSI              0x116
+			MX8MM_IOMUXC_UART2_RXD_ECSPI3_MISO              0x116
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:478]: I think PD is not required.

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 2]
```diff
@@ -282,6 +282,38 @@
 	};
 };
 
+&ecspi3 {
+	fsl,spi-num-chipselects = <6>;
+	cs-gpios = <0>, <&gpio5 1 0>, <&gpio4 25 0>, <&gpio4 26 0>, <&gpio4 23 0>, <&gpio5 3 0>;
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:287]: Something like
```
cs-gpios = <0>,
           <&gpio5  1 GPIO_ACTIVE_HIGH>,
           ...
           <&gpio5  3 GPIO_ACTIVE_HIGH>;
```
would make it easier to read.

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 3]
```diff
@@ -282,6 +282,38 @@
...
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi3>;
+	status = "okay";
+	imu1_accel@0 {
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:291]: Please add new line before

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 4]
```diff
@@ -282,6 +282,38 @@
...
+		reg = <3>;
+	};
+	imu3_accel@4 {
+		compatible = "spidev";
+		reg = <4>;
+	};
+	imu3_gyro@5 {
+		compatible = "spidev";
+		reg = <5>;
+	};
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:314]: Please add newline after each node

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 5]
```diff
@@ -270,6 +270,18 @@
       };
 };
 
+&ecspi2 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi2>;
+	status = "okay";
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:278]: please add newline

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 6]
```diff
@@ -418,6 +430,15 @@
...
 	};
 
+	pinctrl_ecspi2: ecspi2grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_ECSPI2_MISO_ECSPI2_MISO            0x116
+			MX8MM_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI            0x116
+			MX8MM_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK            0x16
+			MX8MM_IOMUXC_ECSPI2_SS0_ECSPI2_SS0              0x116
+               >;
+	};
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:472]: Same as for `ecspi3`

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 7]
```diff
@@ -258,6 +258,18 @@
...
+&ecspi1 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>;
+	status = "okay";
+	inclinometer@0 {
+		compatible = "spidev";
+		reg = <0>;
+      };
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:270]: Spaces instead of tabs

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 8]
```diff
@@ -258,6 +258,18 @@
 	};
 };
 
+&ecspi1 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>;
+	status = "okay";
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:266]: Please add newline

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 9]
```diff
@@ -389,26 +401,35 @@
...
-	
+
+	pinctrl_ecspi1: ecspi1grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_ECSPI1_MISO_ECSPI1_MISO            0x116
+			MX8MM_IOMUXC_ECSPI1_MOSI_ECSPI1_MOSI            0x116
+			MX8MM_IOMUXC_ECSPI1_SCLK_ECSPI1_SCLK            0x116
+			MX8MM_IOMUXC_ECSPI1_SS0_ECSPI1_SS0              0x116
+               >;
+	};
```
Comment (htho-lgs) [inline arch/arm64/boot/dts/freescale/crocodile.dtsi:463]: Same as for `ecspi3`

File: arch/arm64/boot/dts/freescale/crocodile.dtsi [hunk 10]
```diff
@@ -439,6 +471,20 @@
...
+		fsl,pins = <
+			MX8MM_IOMUXC_UART1_RXD_ECSPI3_SCLK              0x116
+			MX8MM_IOMUXC_UART1_TXD_ECSPI3_MOSI              0x116
+			MX8MM_IOMUXC_UART2_RXD_ECSPI3_MISO              0x116
+			MX8MM_IOMUXC_UART2_TXD_ECSPI3_SS0               0x116   /* IMU_1_CS_ACC  */
+			MX8MM_IOMUXC_SAI3_TXD_GPIO5_IO1                 0x116   /* IMU_1_CS_GYRO */
+			MX8MM_IOMUXC_SAI2_TXC_GPIO4_IO25                0x116   /* IMU_2_CS_ACC  */
+			MX8MM_IOMUXC_SAI2_TXD0_GPIO4_IO26               0x116   /* IMU_2_CS_GYRO */
+			MX8MM_IOMUXC_SAI2_RXD0_GPIO4_IO23               0x116   /* IMU_3_CS_ACC  */
+			MX8MM_IOMUXC_SPDIF_TX_GPIO5_IO3                 0x116   /* IMU_3_CS_GYRO */
```
[the remark this answers is not in the document]
Comment (grembeter) [reply arch/arm64/boot/dts/freescale/crocodile.dtsi]: :+1:

File: arch/arm64/boot/dts/freescale/crocodile.dtsi (hunk 1 above)
Comment (grembeter) [reply arch/arm64/boot/dts/freescale/crocodile.dtsi]: :+1:

File: arch/arm64/boot/dts/freescale/crocodile.dtsi (hunk 2 above)
Comment (grembeter) [reply arch/arm64/boot/dts/freescale/crocodile.dtsi]: :+1:

File: arch/arm64/boot/dts/freescale/crocodile.dtsi (hunk 5 above)
Comment (grembeter) [reply arch/arm64/boot/dts/freescale/crocodile.dtsi:279]: :+1:

File: arch/arm64/boot/dts/freescale/crocodile.dtsi (hunk 6 above)
Comment (grembeter) [reply arch/arm64/boot/dts/freescale/crocodile.dtsi]: :+1:

**Pull request merged**